Index: Разное/три точки и прямые.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/Разное/три точки и прямые.py b/Разное/три точки и прямые.py
new file mode 100644
--- /dev/null	(date 1705327068392)
+++ b/Разное/три точки и прямые.py	(date 1705327068392)
@@ -0,0 +1,36 @@
+import tkinter as tk
+
+def draw_line(canvas, x1, y1, x2, y2):
+    canvas.create_line(x1, y1, x2, y2)
+
+root = tk.Tk()
+canvas = tk.Canvas(root, width=400, height=400)
+canvas.pack()
+
+x1, y1 = 10, 10
+x2, y2 = 50, 50
+x3, y3 = 25, 25
+
+k1 = (y2 - y1) / (x2 - x1)
+c1 = y1 - k1 * x1
+
+k2 = -1 / k1
+c2 = y3 - k2 * x3
+
+print(f"Исходная прямая: y = {k1}x + {c1}")
+print(f"Перпендикулярная прямая: y = {k2}x + {c2}")
+
+
+# Конечные точки для исходной прямой
+end_x1 = x1 + 50
+end_y1 = k1 * end_x1 + c1
+
+# Конечные точки для перпендикулярной прямой
+end_x2 = x3 + 20
+end_y2 = k2 * (end_x2 - x3) + y3
+
+# Рисуем прямые на холсте
+draw_line(canvas, x1, y1, end_x1, end_y1)
+draw_line(canvas, x3, y3, end_x2, end_y2)
+
+root.mainloop()
Index: Разное/колич левых поворотов.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/Разное/колич левых поворотов.py b/Разное/колич левых поворотов.py
new file mode 100644
--- /dev/null	(date 1705326904769)
+++ b/Разное/колич левых поворотов.py	(date 1705326904769)
@@ -0,0 +1,20 @@
+def count_left_turns(road):
+    left_turns = 0
+
+    for i in range(len(road) - 2):
+        x1, y1 = road[i]
+        x2, y2 = road[i+1]
+        x3, y3 = road[i+2]
+        cross_product = (x2 - x1) * (y3 - y2) - (y2 - y1) * (x3 - x2)
+
+        if cross_product > 0:
+            left_turns += 1
+
+    return left_turns
+# Определение массива точек
+road = [(0, 0), (1, 2), (2, 1), (4, 4), (5, 2)]
+
+# Вызов функции для подсчета количества левых поворотов
+left_turns = count_left_turns(road)
+
+print(f"Количество левых поворотов: {left_turns}")
\ No newline at end of file
